pkg/agent/core/ngt/service/ngt.go

```diff
@@ -198,26 +210,95 @@ func New(cfg *config.NGT, opts ...Option) (nn NGT, err error) {
 	return n, nil
 }
 
+// migrate migrates the index directory from old to new under the input path if necessary.
+// Migration happens when the path is not empty and there is no `path/origin` directory,
+// which indicates that the user has NOT been using CoW mode and the index directory is not migrated yet.
+func migrate(path string) (err error) {
+	// check if migration is required
+	files, err := file.ListInDir(path)
+	if err != nil {
+		return err
+	}
+	if len(files) == 0 {
+		// empty directory doesn't need migration
+		return nil
+	}
+	originExists := false
+	for _, file := range files {
+		if file == filepath.Join(path, originIndexDirName) {
+			originExists = true
+			break
+		}
+	}
+	if originExists {
+		// origin folder exists. meaning already migrated
+		return nil
+	}
+
+	// at this point, there is something in the path, but there is no `path/origin`, which means migration is required
+	// so create origin and move all contents in path to `path/origin`
+
+	// first move all contents to temporary directory because it's not possible to directly move directory to its subdirectory
+	tp, err := file.MkdirTemp("")
+	if err != nil {
+		return err
+	}
+	err = file.MoveDir(context.Background(), path, tp)
```

Comment:
it would be better to get context from upper layer.

Reply:
[e40e9cb](https://github.com/vdaas/vald/pull/2034/commits/e40e9cb461dd091b9333e227bae8bc26b03a73c8)